go/cli/mcap/cmd/metadata.go

```diff
@@ -88,6 +98,123 @@ var listMetadataCmd = &cobra.Command{
 	},
 }
 
+var addMetadataCmd = &cobra.Command{
+	Use:   "metadata",
+	Short: "Add metadata to an mcap file",
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx := context.Background()
+		if len(args) != 1 {
+			die("Unexpected number of args")
+		}
+		filename := args[0]
+		tempName := filename + ".new"
+		tmpfile, err := os.Create(tempName)
+		if err != nil {
+			die("failed to create temp file: %s", err)
+		}
+		keyvals := make(map[string]string)
+		for _, kv := range addMetadataKeyValues {
+			parts := strings.FieldsFunc(kv, func(c rune) bool {
+				return c == '='
+			})
+			if len(parts) != 2 {
+				die("failed to parse key/value %s", kv)
+			}
+			keyvals[parts[0]] = parts[1]
+		}
+		err = utils.WithReader(ctx, filename, func(remote bool, rs io.ReadSeeker) error {
+			if remote {
+				die("not supported on remote mcap files")
+			}
+			return utils.RewriteMCAP(tmpfile, rs, func(w *mcap.Writer) error {
+				return w.WriteMetadata(&mcap.Metadata{
+					Name:     addMetadataName,
+					Metadata: keyvals,
+				})
+			})
+		})
+		if err != nil {
+			die("failed to add metadata: %s", err)
+		}
+		err = os.Rename(tempName, filename)
+		if err != nil {
+			die("failed to rename temporary output: %s", err)
+		}
+	},
+}
+
+var getMetadataCmd = &cobra.Command{
+	Use:   "metadata",
+	Short: "get metadata by name",
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx := context.Background()
+		if len(args) != 1 {
+			die("Unexpected number of args")
+		}
+		filename := args[0]
+		err := utils.WithReader(ctx, filename, func(_ bool, rs io.ReadSeeker) error {
+			reader, err := mcap.NewReader(rs)
+			if err != nil {
+				return fmt.Errorf("failed to build reader: %w", err)
+			}
+			info, err := reader.Info()
+			if err != nil {
+				return fmt.Errorf("failed to collect mcap info: %w", err)
+			}
+
+			output := make(map[string]string)
+			for _, idx := range info.MetadataIndexes {
+				if idx.Name == getMetadataName {
+					_, err := rs.Seek(int64(idx.Offset+1+8), io.SeekStart)
+					if err != nil {
+						return fmt.Errorf("failed to seek to metadata record at %d: %w", idx.Offset, err)
+					}
+					data := make([]byte, idx.Length)
+					_, err = io.ReadFull(rs, data)
+					if err != nil {
+						return fmt.Errorf("failed to read metadata record: %w", err)
+					}
+					record, err := mcap.ParseMetadata(data)
+					if err != nil {
+						return fmt.Errorf("failed to parse metadata: %w", err)
+					}
+					for k, v := range record.Metadata {
+						output[k] = v
+					}
+				}
+			}
+			jsonBytes, err := json.Marshal(output)
+			if err != nil {
+				return fmt.Errorf("failed to marshal output to JSON: %w", err)
+			}
+			buf := &bytes.Buffer{}
+			err = json.Indent(buf, jsonBytes, "", "    ")
+			if err != nil {
+				return fmt.Errorf("failed to format metadata: %w", err)
+			}
+			_, err = buf.Write([]byte{'\n'})
+			if err != nil {
+				return fmt.Errorf("failed to write terminal newline: %w", err)
+			}
```

Comment:
we can pull this indenting + newline termination into a json pretty print function